File: dummy-ai-client/NetworkStandard/DataTypes/Gadgets/wiretap_with_earplugs.py
from NetworkStandard.DataTypes.Gadgets.gadget import Gadget
from NetworkStandard.DataTypes.gadget_enum import GadgetEnum

import uuid


class WiretapWithEarplugs(Gadget):

    def __init__(self, gadget: GadgetEnum):
        super.__init__(gadget)
        self.working = False;
        self.active_on = None

    def set_working(self, working):
        self.working = working

    def set_active_on(self, active_on: uuid.UUID):
        self.active_on = active_on

File: backend/src/main/java/NetworkStandard/DataTypes/Gadgets/WiretapWithEarplugs.java
package NetworkStandard.DataTypes.Gadgets;

import NetworkStandard.DataTypes.GadgetEnum;

import java.util.UUID;
/**
 * @author Marios Sirtmatsis
 */
public class WiretapWithEarplugs extends Gadget {
    private Boolean working;    //describes whether the gadget still works
    private UUID activeOn;      //desribes on which target the gadget ist active. If the gadget isn't activated, set the field to null.
    private double wireTapProbability;
    private boolean isWiretap;

    /**
     * Constructor.
     *
     * @param gadget - type of gadget
     */
    public WiretapWithEarplugs(GadgetEnum gadget) {
        super(gadget);
    }

    public double getWireTapProbability() {
        return wireTapProbability;
    }

    public void setWireTapProbability(double wireTapProbability) {
        this.wireTapProbability = wireTapProbability;
    }

    public Boolean getWorking() {
        return working;
    }

    public UUID getActiveOn() {
        return activeOn;
    }

    public void setWorking(Boolean working) {
        this.working = working;
    }

    public void setActiveOn(UUID activeOn) {
        this.activeOn = activeOn;
    }

    public boolean isWiretap() {
        return isWiretap;
    }

    public void setWiretap(boolean wiretap) {
        isWiretap = wiretap;
    }
}
